py/server/deephaven/liveness_scope.py

```diff
@@ -8,11 +8,34 @@
 
 import jpy
 
+from typing import Union
+from warnings import warn
+
 from deephaven import DHError
 from deephaven._wrapper import JObjectWrapper
 
 _JLivenessScopeStack = jpy.get_type("io.deephaven.engine.liveness.LivenessScopeStack")
 _JLivenessScope = jpy.get_type("io.deephaven.engine.liveness.LivenessScope")
+_JLivenessReferent = jpy.get_type("io.deephaven.engine.liveness.LivenessReferent")
+
+
+def _push(scope: _JLivenessScope):
+    _JLivenessScopeStack.push(scope)
+
+
+def _pop(scope: _JLivenessScope):
+    try:
+        _JLivenessScopeStack.pop(scope)
+    except Exception as e:
+        raise DHError(e, message="failed to pop the LivenessScope from the stack.")
+
+
+def _unwrap_to_liveness_referent(referent: Union[JObjectWrapper, jpy.JType]) -> jpy.JType:
+    if isinstance(referent, jpy.JType) and _JLivenessReferent.jclass.isInstance(referent):
+        return referent
+    if isinstance(referent, JObjectWrapper):
+        _unwrap_to_liveness_referent(referent.j_object)
```

Comment:
```suggestion
        return _unwrap_to_liveness_referent(referent.j_object)
```

Found by fixing tests - so far at least the other test failures have to do with the bad patterns that we forced on users (including our own tests).